File: generated_code/optimized/memory/java/0.java
import java.util.*;
import java.lang.*;

class Solution {
    public boolean hasCloseElements(List<Double> numbers, double threshold) {
        int size = numbers.size();
        for (int i = 0; i < size - 1; i++) {
            double a = numbers.get(i);
            for (int j = i + 1; j < size; j++) {
                if (Math.abs(a - numbers.get(j)) < threshold) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: generated_code/optimized/balanced/detailed/cpp/0.cpp
#include <vector>
#include <algorithm>

using namespace std;

bool has_close_elements(vector<float> numbers, float threshold) {
    const size_t n = numbers.size();
    if (n < 2 || threshold <= 0.0f) return false;

    sort(numbers.begin(), numbers.end());

    for (size_t i = 1; i < n; ++i) {
        if (numbers[i] - numbers[i - 1] < threshold) {
            return true;
        }
    }

    return false;
}
